woocommerce/woocommerce
# Add introduction banner card in marketing page

### All Submissions:


### Changes proposed in this Pull Request:

Part of https://github.com/woocommerce/woocommerce/issues/34904.

Note: This PR depends on PR https://github.com/woocommerce/woocommerce/pull/36130, to display the "create a campaign" modal upon clicking the "Create a campaign" button. Since PR https://github.com/woocommerce/woocommerce/pull/36130 is being put aside for now, this PR would be put on hold too.

In this PR, we add the introduction banner card into multichannel marketing page.

There are two "types" of introduction banner card: one with the "Create a campaign" and "Add channels" buttons, and one without.

Screenshot of introduction banner card without the "Create a campaign" and "Add channels" buttons - this will be shown for users who has no channels setup yet:

![image](https://user-images.githubusercontent.com/417342/209806777-a220e7f2-8937-45b2-bc07-726a2d0a2b19.png)

Screenshot of introduction banner card with the "Create a campaign" and "Add channels" buttons - this will be shown when users has at least one channel:

![image](https://user-images.githubusercontent.com/417342/209806644-dad37194-5224-4f60-98b0-353c9d1eb7a8.png)

Users can click on the "X" button at the top right corner of the card to dismiss the banner. We use a new WP Option `woocommerce_marketing_overview_multichannel_banner_dismissed` to store the dismissal state. This approach is similar to the existing [`woocommerce_marketing_overview_welcome_hidden`](https://github.com/woocommerce/woocommerce/blob/3daeac9a698f71f84619a37335d5289598a5b394/plugins/woocommerce-admin/client/marketing/overview/welcome-card/index.js#L21-L23) WP Option for the WelcomeCard in the existing Marketing page.

### How to test the changes in this Pull Request:

Pre-conditions: 

- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features page: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
- You should not have any channels or campaigns setup before.
- You should not have the `woocommerce_marketing_overview_multichannel_banner_dismissed` option in the wp_options database table.

Test steps:

1. Go to WooCommerce Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
2. You should see the introduction banner card without the "Create a campaign" and "Add channels" buttons like the first screenshot above.
3. Proceed to setup a channel. 
4. Refresh the page. You should see the introduction banner card with the "Create a campaign" and "Add channels" buttons like the second screenshot above.

### Other information:


### FOR PR REVIEWER ONLY:
